src/api/app/models/project.rb

```diff
@@ -251,6 +258,49 @@ def cleanup_linking_targets
     end
   end
 
+  def update_meta(request_data, params)
+    result = false
+
+    Project.check_target_of_link(name, request_data)
+    Project.update_project_meta_maintains(request_data)
+    new_repo_names = Project.update_project_meta_repositories(self.name, request_data)
+
+    # permission check passed, write meta
+    removedRepositories = []
+    self.repositories.each do |repo|
+      if !new_repo_names[repo.name] && !repo.remote_project_name
+        removedRepositories << repo
+      end
+    end
+
+    return result unless check_and_remove_repositories(params, removedRepositories)
+
+    Project.transaction do
+      self.update_from_xml(request_data)
+      self.store
+      result = true
+    end
+    result
```

Comment:
Not 100% sure, but I think this will never return false. Either it's true or there is an exception